### ISSUE TITLE
Start-up errors?

### ISSUE THREAD

**Issue by OksmaWarez** [canyonmodded/canyon#58]
`10:49:49 [SEVERE] java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
10:49:49 [SEVERE]       at net.minecraft.server.RegionFile.<init>(RegionFile.java:77)
10:49:49 [SEVERE]       at net.minecraft.server.RegionFileCache.a(RegionFileCache.java:40)
10:49:49 [SEVERE]       at net.minecraft.server.RegionFileCache.c(RegionFileCache.java:72)
10:49:49 [SEVERE]       at net.minecraft.server.ChunkRegionLoader.a(ChunkRegionLoader.java:19)
10:49:49 [SEVERE]       at net.minecraft.server.ChunkProviderServer.realLoadChunk(ChunkProviderServer.java:188)
10:49:49 [SEVERE]       at net.minecraft.server.ChunkProviderServer.requestChunkLoad(ChunkProviderServer.java:331)
10:49:49 [SEVERE]       at net.minecraft.server.ChunkProviderServer.getChunkAtAsync(ChunkProviderServer.java:94)
10:49:49 [SEVERE]       at net.minecraft.server.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:157)
10:49:49 [SEVERE]       at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:352)
10:49:49 [SEVERE]       at net.minecraft.server.MinecraftServer.init(MinecraftServer.java:239)
10:49:49 [SEVERE]       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:432)
10:49:49 [SEVERE]       at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
10:49:49 [SEVERE]       at java.lang.Thread.run(Thread.java:748)
10:49:49 [SEVERE] Unexpected exception
java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
        at net.minecraft.server.RegionFile.<init>(RegionFile.java:77)
        at net.minecraft.server.RegionFileCache.a(RegionFileCache.java:40)
        at net.minecraft.server.RegionFileCache.c(RegionFileCache.java:72)
        at net.minecraft.server.ChunkRegionLoader.a(ChunkRegionLoader.java:19)
        at net.minecraft.server.ChunkProviderServer.realLoadChunk(ChunkProviderServer.java:188)
        at net.minecraft.server.ChunkProviderServer.requestChunkLoad(ChunkProviderServer.java:331)
        at net.minecraft.server.ChunkProviderServer.getChunkAtAsync(ChunkProviderServer.java:94)
        at net.minecraft.server.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:157)
        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:352)
        at net.minecraft.server.MinecraftServer.init(MinecraftServer.java:239)
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:432)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)`
It still starts the server tho?

**Comment by KoboDev** [comment]
Is this the full log?
What Java version are you running?

**Comment by OksmaWarez** [comment]
I'm using Java 8.
```[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
[Pterodactyl Daemon]: Finished pulling Docker container image
                                                                      
 mmmmmm        ""#      "           mm   m            #               
 #       mmm     #    mmm    m   m  #"m  #  mmm    mmm#   mmm    mmm  
 #mmmmm "   #    #      #     #m#   # #m # #" "#  #" "#  #"  #  #   " 
 #      m"""#    #      #     m#m   #  # # #   #  #   #  #""""   """m 
 #      "mm"#    "mm  mm#mm  m" "m  #   ## "#m#"  "#m##  "#mm"  "mmm" 
                                                                      
                                                                      
Found server.jar
===============================
Java 8
===============================
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (Zulu 8.52.0.23-CA-linux64) (build 1.8.0_282-b08)
OpenJDK 64-Bit Server VM (Zulu 8.52.0.23-CA-linux64) (build 25.282-b08, mixed mode)
151 recipes
16 achievements
10:41:00 [INFO] Starting minecraft server version Beta 1.7.3
10:41:00 [INFO] Loading properties
10:41:00 [WARNING] **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
10:41:00 [WARNING] The server will make no attempt to authenticate usernames. Beware.
10:41:00 [WARNING] While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
10:41:00 [WARNING] To change this, set "online-mode" to "true" in the server.settings file. Note that a client mod is required for this to work.
10:41:00 [INFO] This server is running Canyon version git-Canyon-35548c6 (MC: 1.7.3)
10:41:01 [INFO] Starting Minecraft server on 0.0.0.0:58253
10:41:01 [INFO] Preparing level "world"
10:41:01 [INFO] Preparing start region for level 0 (Seed: -8073127223417069985)
10:41:02 [SEVERE] java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
10:41:02 [SEVERE]       at net.minecraft.server.RegionFile.<init>(RegionFile.java:77)
10:41:02 [SEVERE]       at net.minecraft.server.RegionFileCache.a(RegionFileCache.java:40)
10:41:02 [SEVERE]       at net.minecraft.server.RegionFileCache.c(RegionFileCache.java:72)
10:41:02 [SEVERE]       at net.minecraft.server.ChunkRegionLoader.a(ChunkRegionLoader.java:19)
10:41:02 [SEVERE]       at net.minecraft.server.ChunkProviderServer.realLoadChunk(ChunkProviderServer.java:188)
10:41:02 [SEVERE]       at net.minecraft.server.ChunkProviderServer.requestChunkLoad(ChunkProviderServer.java:331)
10:41:02 [SEVERE]       at net.minecraft.server.ChunkProviderServer.getChunkAtAsync(ChunkProviderServer.java:94)
10:41:02 [SEVERE]       at net.minecraft.server.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:157)
10:41:02 [SEVERE]       at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:352)
10:41:02 [SEVERE]       at net.minecraft.server.MinecraftServer.init(MinecraftServer.java:239)
10:41:02 [SEVERE]       at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:432)
10:41:02 [SEVERE]       at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
10:41:02 [SEVERE]       at java.lang.Thread.run(Thread.java:748)
10:41:02 [SEVERE] Unexpected exception
java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
        at net.minecraft.server.RegionFile.<init>(RegionFile.java:77)
        at net.minecraft.server.RegionFileCache.a(RegionFileCache.java:40)
        at net.minecraft.server.RegionFileCache.c(RegionFileCache.java:72)
        at net.minecraft.server.ChunkRegionLoader.a(ChunkRegionLoader.java:19)
        at net.minecraft.server.ChunkProviderServer.realLoadChunk(ChunkProviderServer.java:188)
        at net.minecraft.server.ChunkProviderServer.requestChunkLoad(ChunkProviderServer.java:331)
        at net.minecraft.server.ChunkProviderServer.getChunkAtAsync(ChunkProviderSer
```
This is the full log

**Comment by KoboDev** [comment]
Try an later java version like java9 or 11 jdk. 
It could've been because you compiled using an later JDK but used JDK8, or it could've been because of rewrite of entire code to fix the load of errors and dependency issues. #44

**Comment by 1-Mythical** [comment]
I know I am a bit late here. For anyone who doesn't know what to change here, you need to launch your server from the same Java version used when building with Maven or edit the environment variables to use whatever Java 8 installation you plan to launch the server. Then just build the project with Maven and then run the server jar file in the target folder.

**Comment by OksmaWarez** [comment]
This is a pre-built version

**Comment by KoboDev** [comment]
pre-builts were using java9 and later, sorry.

**Comment by astei** [comment]
This is fixed with 8e5526744a67c64c07ed5a5a0567b71fa29e9bf6
